lib/src/widgets/atomic/report_dialogue.dart

```diff
@@ -0,0 +1,131 @@
+import "package:btc_market/services.dart";
+import "package:btc_market/src/data/report.dart";
+import "package:flutter/material.dart";
+
+/// A report form.  
+/// Should be used as the content of an AlertDialogue.  
+class ReportDialogue extends StatefulWidget {
+  /// The type of item being reported
+  final ItemType itemType;
+
+  /// The ID of the item being reported
+  final String itemID;
+
+  /// Constructs a new report dialogue widget
+  const ReportDialogue({required this.itemType, required this.itemID, super.key});
+
+  @override
+  State<ReportDialogue> createState() => _ReportDialogueState();
+}
+
+/// A bunch of reasons that will appear on the dropdown menu,
+final reportReasons = [
+  "Other",
+  "Inappropriate/NSFW",
+];
+
+/// These ones depends on the type of item being reported
+final specificReportReasons = [
```

Comment:
Good idea, but I think these three concepts specifically can be used for all types of reports (a fraudulent product could be a pencil for $1,000, or a gaming PC for $5). So maybe just merge these with `reportReasons`, and maybe move that to be a static member on `Report` itself: 

```dart
class Report {
  static final possibleReasons= [
    "Innappropriate", 
    // ...
    "Other",
  ];
  // ...
}
```

Then you can refer to `Report.possibleReasons` from any file